doc/configuration.rst

```diff
@@ -188,6 +188,22 @@ the converse does not hold.
 If you so desire you can implement your own sorting key. It will be
 provided the relative paths to `conf.py` of each sub gallery folder.
 
+.. warning:: If you create your own class for ``'subsection_order'``, ensure
+             that the ``__str__`` of your class is stable across runs.
+             Sphinx determines if the build environment has changed
+             (and thus if *all* documents should be rewritten)
+             by examining the config values using
+             ``md5(str(obj).encode()).hexdigest()`` in
+             ``sphinx/builders/html.py``. Default class instances
+             in Python have their memory address in their ``__repr__`` which
+             will in general change for each build. For ``ExplicitOrder``
+             for example, this is fixed via::
+
+                 def __repr__(self):
+                     return '<%s: %s>' % (self.__class__.__name__, self.ordered_list)
+
+             Thus the files are only all rebuilt if the specified ordered list
+             is changed.
```

Comment:
Should downstream packages just subclass `ExplicitOrder`?

Reply:
`mpl` does already

Reply:
Right, I just meant it could be explicitly mentioned here; i.e. either subclass `ExplicitOrder` or make sure you have this `__repr__`....

Reply:
I don't want to suggest people subclass `ExplicitOrder` to get this functionality; and by the same token, people who subclass it for the right reasons (using its intended purpose) should immediately understand that they'll get `__repr__` for free by subclassing.